plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/p5edges/orthogonal/direction/SouthToNorthRoutingStrategy.java

```diff
@@ -0,0 +1,93 @@
+    }
+
+    @Override
+    public PortSide getSourcePortSide() {
+        return PortSide.NORTH;
+    }
+
+    @Override
+    public PortSide getTargetPortSide() {
+        return PortSide.SOUTH;
+    }
+
+    @Override
+    public void calculateBendPoints(final HyperEdgeSegment segment, final double startPos, final double edgeSpacing) {
+        // We don't do anything with dummy segments; they are dealt with when their partner is processed
+        if (segment.isDummy()) {
+            return;
+        }
+        
+        // Calculate coordinates for each port's bend points
+        double segmentY = startPos - segment.getRoutingSlot() * edgeSpacing;
+
+        for (LPort port : segment.getPorts()) {
+            double sourceX = port.getAbsoluteAnchor().x;
+
+            for (LEdge edge : port.getOutgoingEdges()) {
+                if (!edge.isSelfLoop()) {
+                    LPort target = edge.getTarget();
+                    double targetX = target.getAbsoluteAnchor().x;
+                    
+                    if (Math.abs(sourceX - targetX) > OrthogonalRoutingGenerator.TOLERANCE) {
+                        // We'll update these if we find that the segment was split
+                        double currentY = segmentY;
+                        HyperEdgeSegment currentSegment = segment;
+
+                        KVector bend = new KVector(sourceX, currentY);
+                        edge.getBendPoints().add(bend);
+                        addJunctionPointIfNecessary(edge, currentSegment, bend, false);
+
+                        // If this segment was split, we need two additional bend points
+                        HyperEdgeSegment splitPartner = segment.getSplitPartner();
+                        if (splitPartner != null) {
+                            double splitX = splitPartner.getIncomingConnectionCoordinates().get(0);
+                            
+                            bend = new KVector(splitX, currentY);
+                            edge.getBendPoints().add(bend);
+                            addJunctionPointIfNecessary(edge, currentSegment, bend, false);
+                            
+                            // Advance to the split partner's routing slot
+                            currentY = startPos - splitPartner.getRoutingSlot() * edgeSpacing;
+                            currentSegment = splitPartner;
+                            
+                            bend = new KVector(splitX, currentY);
+                            edge.getBendPoints().add(bend);
+                            addJunctionPointIfNecessary(edge, currentSegment, bend, false);
+                        }
+
+                        bend = new KVector(targetX, currentY);
+                        edge.getBendPoints().add(bend);
+                        addJunctionPointIfNecessary(edge, currentSegment, bend, false);
+                    }
+                }
+            }
+        }
+    }
+    
```

Comment:
Remove empty line

Reply:
I like to have an empty line between class content and the class's block statement.